app/controllers/application_controller.rb

```diff
@@ -1,6 +1,7 @@
 class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
+  include Pundit
   protect_from_forgery with: :exception
   before_action :authenticate_user!
 end
```

Comment:
So the reason your extra user fields are not persisting is because you didn't configure sanitized parameters for devise. Check out this article on how to do that:

http://stackoverflow.com/questions/19791531/how-to-specify-devise-parameter-sanitizer-for-edit-action